src/build/loaders/json-parts-loader.js

```diff
@@ -0,0 +1,32 @@
+const path = require("path");
+
+/**
+ * @param {string} jsonFileContent
+ * @param {{ importPart: (filename: string) => string | Promise<string> }} options
+ * @returns {Promise<string>} resulting js module
+ */
+async function compileJsonParts(jsonFileContent, { importPart }) {
+  /** @type {string[]} */
+  const partPaths = JSON.parse(jsonFileContent);
+  const parts = await Promise.all(
+    partPaths.map((filename) => importPart(filename))
+  );
+  const content = parts.join("<!-- part separator -->\n");
```

Comment:
Maybe this marker can be an exported constant that the revealjs loader can import to split?